### ISSUE TITLE
Live documentation vs released code confusion

### ISSUE THREAD

**Issue by bruceg** [timberio/vector#2649]
TL;DR The documentation on the web site is newer than the released version of Vector. This has caused repeated confusion and needs to be resolved.

The process that generates the [documentation website](https://vector.dev/docs/) is triggered on every commit to master. This means that the live documentation will always track the latest code we merge in. Most users, however, are running a stable release rather than updating to each nightly release (technically, even the nightly releases are behind the documentation, but only by less than 24 hours).

The discrepancies between our live documentation and released code has caused a number of confused users that seems to be increasing as Vector grows in popularity. This may be a good sign for a measure of our growth, but is a negative reflection on our public image in general. We should figure some way of resolving this confusion to prevent it from happening again.

I see two possibilities for how to handle this, roughly speaking "internal" and "external".

1. Ensure that every feature's documentation marks either its introduction or its modification (for things like breaking changes, etc) within the documentation. I don't foresee a way of doing this with an easily automated tool, so this means additional work for each externally visible change. Additionally, it introduces complications to the release process, as the version number in which a change was introduced may not be known until the actual point of release. On the other hand, it makes for a great reference point on every page of documentation for the advances in capabilities of Vector.

2. Introduce separate top-level documentation pages on the web site. The default link would point to the stable documentation (updated only when we actually tag a release), with additional an additional link to nightly documentation. This has the advantage of requiring no developer work once the system is set up, but makes comparisons between versions more difficult.

**Comment by lukesteensen** [comment]
I agree this is a big problem and we should try to address it as soon as possible.

The simplest thing we could do is probably to turn off auto-publishing on each push and run a publish command as part of the release process.

Longer-term, it may be possible to take advantage of netlify's redirects and ability to deploy branches to get more fully-featured versioned documentation. I believe @binarylogic has looked into this.

**Comment by Jeffail** [comment]
I had this issue with benthos and chose to maintain a separate branch that deploys the docs site, each time you release you rebase the docs branch to the tag. The main motivation for that is it allows you to occasionally cherry-pick docs fixes from master that are relevant to the current release whilst omitting docs for new features.

**Comment by binarylogic** [comment]
I've updated the default branch on netlify to point to `v0.9`. This is a manual step that we can automate as part of the release process (https://github.com/timberio/vector/issues/2863). As such, I'm closing this issue.
